core/src/com/unciv/logic/automation/UnitAutomation.kt

```diff
@@ -249,7 +249,12 @@ object UnitAutomation {
         val closestEnemy = closeEnemies.minBy { it.tileToAttack.aerialDistanceTo(unit.getTile()) }
 
         if (closestEnemy != null) {
-            unit.movement.headTowards(closestEnemy.tileToAttackFrom)
+            val tileOfLeastInfuence = closestEnemy.tileToAttackFrom.getTilesAtDistance(1)
```

Comment:
You won't necessarily be able to attack from there. I'm not sure what you're aiming for here, to be honest.

Reply:
So I fixed that issue about checking if the movement is valid. 

My thought is that if the unit can't attack this turn it might as well go to a tile that has the most friendly influence, then attack next turn. That way unit doesn't stupidly charge in, it'll be grouped together with other friendly units or move to terrain that is better defendable.